crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -358,6 +387,10 @@ export.finalize_transaction
     hmerge
     # => [ACCOUNT_UPDATE_COMMITMENT]
 
+    # load fee asset from local
+    padw loc_loadw.0
+    # => [FEE_ASSET, ACCOUNT_UPDATE_COMMITMENT]
```

Comment:
nit: it took me a bit of time to realize that we are putting the `FEE_ASSET` onto the stack here to pass it as a parameter to the `build_output_vault` procedure. I would expand the comment a bit to mention that this basically prepares the stack for `build_output_vault`.